### PR TITLE
frontend-ui: docs and folders scaffolding

### PR DESCRIPTION
**Pull request by murphp15**
# What
We want to do preliminary work on setting up folder structure/docs etc for the frontend code. 

# Why 
It will be a first step in closing this https://github.com/vmware/versatile-data-kit/issues/1507. 

# How has this been tested 
No testing necessary  

- [x] add .gitignore, and initially empty .gitlab-ci/ dir
- [ ]  remove .reviewboardrc and all from support/* -> except eventually keep open_docker.ps1
- [x]  we should keep the CHANGELOG.md, rename header, and move it and version.txt to 3.0.0 (keep the all exact entries including @vmw-prefixed as well)
- [x] cleanup the CHANGELOG.md (in OSS, changelog is dynamically generated)
- [x] move projects/gui/version.txt to 3.0.0
- [x] cleanup README.md from internal information
- [x] rework the .md documentation to workaround internal links, instead link to the VEP and make sure to transfer any needed information there. 
- [x] projects/gui/proxy.config.jsonrequires a data pipelines control service instance for e2e testing -> reuse http://cicd-control-service-svc:8092/ k8s-internal domain like the vdk-heartbeat
- [ ] cleanup metadata service entries
- [x] keep http://localhost.vmware.com:4200/
- [x] Take a look what directories to keep, needed for old @taurus/data-pipelines:[^taurus_shared_open_source_plan.txt], a part of the research story https://jira.eng.vmware.com/browse/SC-1656
- [x] e2e tests are not data-pipelines -related, so all e2e/ dir content is descoped

Signed-off-by: murphp15 <murphp15@tcd.ie>

### CODE REVIEW

File: projects/frontend-ui/shared-components/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# VDK Shared Components library
+
+This project implements common UI functionality that is used in the data-pipelines project.
+But these components are also used by private projects within VMWare.
```
Comment (ivakoleva) [inline projects/frontend-ui/shared-components/README.md:4]: I would rephrase to emphasise these components are agnostic (non data-pipelines UI -specific), and could be used for extension with other GUI components

File: projects/frontend-ui/shared-components/.gitignore [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.idea/
+projects/gui/debug.log
+projects/gui/e2e/hars/
+projects/gui/e2e/logs/
+projects/gui/e2e/reports/
+projects/gui/e2e/screenshots/
+projects/gui/e2e/videos/
+projects/gui/reports/
+.DS_Store
```
Comment (ivakoleva) [inline projects/frontend-ui/shared-components/.gitignore:9]: maybe reflect the directories we currently have, for example `projects/gui/e2e` would not have contents for now or should not exist at all, no reason to have this git-ignored

File: projects/frontend-ui/shared-components/projects/gui/README.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+   ```bash
+   npm link
+   ```
+5. Return to project dir
+   ```bash
+   cd ../../
+   ```
+6. Create symlink from build artifact to the wrapper ui (documentation-ui)
+   ```bash
+   npm link @taurus/shared
```
Comment (ivakoleva) [inline projects/frontend-ui/shared-components/projects/gui/README.md:50]: what is "taurus"? should be refactored

File: projects/frontend-ui/shared-components/projects/gui/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+The Shared Components contains:
+
+* **[Shared Components Library](projects/shared/README.md)** Library that holds UI Components.
+* **[Documentation UI](projects/documentation-ui/src)** Reference UI Wrapper and Documentation Implementation
+
+
+## Running and Testing
+
+You can use the reference implementation in `projects/documentation-ui` as showcase of the library.
```
Comment (ivakoleva) [inline projects/frontend-ui/shared-components/projects/gui/README.md:9]: not sure yet about documentation-ui, my current understanding is we go without documentation-ui, so lets remove it for now

File: projects/frontend-ui/shared-components/projects/gui/README.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+### Running tests
+
+[Unit] Run `npm run test` to execute the unit tests via [Karma](https://karma-runner.github.io).
+
+_**Note**_: Code coverage report will be generated in `reports/coverage`(for the UI Document Wrapper and the Shared Components Library)
+
+_**Note**_: Code coverage report for the Shared UI Components library will be also logged in the Console(and consumed by the CI/CD)
+
+[E2E] Run `npm run e2e` to open [Cypress](https://www.cypress.io/) dev tool for e2e testing
```
Comment (ivakoleva) [inline projects/frontend-ui/shared-components/projects/gui/README.md:68]: no data-pipelines -related e2e tests as agreed, we should drop this

File: projects/frontend-ui/shared-components/projects/gui/projects/shared/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
+# VDK Shared Components library
+
+This project implements common UI functionality that is used in the data-pipelines project.
+But these components are also used by private projects within VMWare.
```
Comment (ivakoleva) [inline projects/frontend-ui/shared-components/projects/gui/projects/shared/README.md:4]: this is duplicated with the other readme / redundant?

File: projects/frontend-ui/shared-components/projects/gui/proxy.config.json [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  "/metadata/lineage/datajob:test_datajob": {
+    "target": "http://localhost.vmware.com:4200/assets/lineage-data/lineage_result.json",
+    "pathRewrite": {
+      "^/metadata/lineage/datajob:test_datajob": ""
+    },
+    "changeOrigin": true,
+    "secure": false,
+    "logLevel": "debug"
+  },
+  "/metadata": {
```
Comment (ivakoleva) [inline projects/frontend-ui/shared-components/projects/gui/proxy.config.json:11]: remove all metadata-related

File: specs/vep-1507-vdk-operations-ui/README.md [hunk 8]
```diff
@@ -48,6 +48,12 @@ One example is features that were cut during scoping.
 
 ## High-level design
 
+#### Folder Structure
+
+* [frontend-ui](../../projects/frontend-ui) The root folder for all frontend code.
+  * [shared-components](../../projects/frontend-ui/shared-components/README.md): the root folder for the shared components project, for more details please see the readme.
```
Comment (ivakoleva) [inline specs/vep-1507-vdk-operations-ui/README.md:54]: suggestion: drop "-components" suffix, since a loot of directories are components so -components is actually redundant.

File: specs/NNNN-template/README.md [hunk 9]
```diff
@@ -4,12 +4,6 @@
 * **Author(s):** Name (mail), ...
 * **Status:** draft | implementable | implemented | rejected | withdrawn | replaced
 
-<!--
-**Note:** Any PRs to move a VEP to `implementable`, or significant changes once
-it is marked `implementable`, must be approved by at least 2 maintainers .
-
--->
-
```
Comment (ivakoleva) [inline specs/NNNN-template/README.md:12]: Why editing the VEP template itself?

File: specs/vep-1507-vdk-operations-ui/README.md [hunk 10]
```diff
@@ -48,6 +48,12 @@ One example is features that were cut during scoping.
```
Comment (ivakoleva) [inline specs/vep-1507-vdk-operations-ui/README.md:48]: Please elaborate on what frontend-ui is?

File: projects/frontend-ui/shared-components/projects/gui/README.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+0. Set npm registry to use
+   * There is .npmrc file in the project that refers to correct registries and provide configuration
+   * If something is not working with autoconfiguration, manually set registries
+
+   ```bash
+   $ npm config set legacy-peer-deps true
+   ```
+1. Install Angular Cli and all other dependencies except '@taurus/shared'
+   ```bash
+   npm i -g @angular/cli
```
Comment (ivakoleva) [inline projects/frontend-ui/shared-components/projects/gui/README.md:27]: isn't there a specific version compatibility? we should add this, and also describe it in the VEP

File: projects/frontend-ui/shared-components/README.md (hunk 1 above)
Comment (murphp15) [reply projects/frontend-ui/shared-components/README.md]: Done

File: projects/frontend-ui/shared-components/projects/gui/README.md (hunk 11 above)
Comment (murphp15) [reply projects/frontend-ui/shared-components/projects/gui/README.md]: @ivakoleva  why do you think there is a specific version compatibly? 
I just run this command as described and everything worked correctly.

File: specs/vep-1507-vdk-operations-ui/README.md (hunk 10 above)
Comment (murphp15) [reply specs/vep-1507-vdk-operations-ui/README.md:48]: I was going to do this as part of the data-pipelines PR. because that folder is more focused on the actual functionality. 
WDYT?

File: projects/frontend-ui/shared-components/projects/gui/projects/shared/README.md (hunk 6 above)
Comment (murphp15) [reply projects/frontend-ui/shared-components/projects/gui/projects/shared/README.md]: Removed. 
This duplication kinda existed in the source repo.

File: projects/frontend-ui/shared-components/projects/gui/README.md (hunk 4 above)
Comment (murphp15) [reply projects/frontend-ui/shared-components/projects/gui/README.md]: removed all references to documentation UI.

File: specs/vep-1507-vdk-operations-ui/README.md (hunk 10 above)
Comment (ivakoleva) [reply specs/vep-1507-vdk-operations-ui/README.md:48]: you are mentioning the data-pipelines and the readme already, and it makes sense to either explain it now since mentioned, or move the data-pipelines-related changes entirely into a separate PR

File: projects/frontend-ui/shared/projects/gui/proxy.config.json [hunk 12]
```diff
@@ -0,0 +1,11 @@
+{
+  "/metadata/lineage/datajob:test_datajob": {
```
Comment (ivakoleva) [inline projects/frontend-ui/shared/projects/gui/proxy.config.json:2]: metadata" should not be a part of the PR

File: projects/frontend-ui/shared/CHANGELOG.md [hunk 13]
```diff
@@ -0,0 +1,2 @@
+Change Log - VDK Shared Components Library (@vdk/shared)
+=========
```
Comment (ivakoleva) [inline projects/frontend-ui/shared/CHANGELOG.md:2]: do we use CHANGELOG.md anywhere else in versatile-data-kit, or is it a dynamic one on new release?

File: projects/frontend-ui/shared/README.md [hunk 14]
```diff
@@ -0,0 +1,17 @@
+# VDK Shared Components library
+
+This project implements common UI functionality that is used in the data-pipelines project.
+Everything in this folder is project agnostic and there is nothing data-pipelines specific about them.
+The hope with having them in a separate directory is that other projects might leverage them.
+At the moment there are many other private projects in VMware which are using them.
```
Comment (ivakoleva) [inline projects/frontend-ui/shared/README.md:6]: Don't think it's needed to mention who uses it if not some public artifacts, it's out of scope of the public repository;

File: projects/frontend-ui/shared/README.md [hunk 15]
```diff
@@ -0,0 +1,17 @@
+# VDK Shared Components library
+
+This project implements common UI functionality that is used in the data-pipelines project.
+Everything in this folder is project agnostic and there is nothing data-pipelines specific about them.
```
Comment (ivakoleva) [inline projects/frontend-ui/shared/README.md:4]: the "shared" shouldn't know about data-pipelines (since agnostic), I would even keep that out of the shared README. It's just meant to be reused.

File: specs/vep-1507-vdk-operations-ui/README.md (hunk 8 above)
Comment (ivakoleva) [reply specs/vep-1507-vdk-operations-ui/README.md]: or maybe shared-components is good; it was like that before, right?

File: projects/frontend-ui/shared/projects/gui/proxy.config.json (hunk 12 above)
Comment (murphp15) [reply projects/frontend-ui/shared/projects/gui/proxy.config.json:2]: Will I delete the entire project.config.json file?

File: projects/frontend-ui/shared/README.md (hunk 14 above)
Comment (murphp15) [reply projects/frontend-ui/shared/README.md:6]: I think it give legitimacy to the decision. But happy to remove it also.

File: projects/frontend-ui/shared/projects/gui/README.md [hunk 16]
```diff
@@ -0,0 +1,66 @@
+The Shared Components contains:
+
+* **[Shared Components Library](projects/shared/README.md)** Library that holds UI Components.
+* **[Documentation UI](projects/documentation-ui/src)** Reference UI Wrapper and Documentation Implementation
```
[the remark this answers is not in the document]
Comment (murphp15) [reply projects/frontend-ui/shared/projects/gui/README.md]: This is removed now. 
I actually had it removed locally put the push failed. 
Sorry about that.

File: projects/frontend-ui/shared/projects/gui/README.md [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+0. Set npm registry to use
+   * There is .npmrc file in the project that refers to correct registries and provide configuration
+   * If something is not working with autoconfiguration, manually set registries
+
+   ```bash
+   $ npm config set legacy-peer-deps true
+   ```
+1. Install Angular Cli and all other dependencies except '@vdk/shared'
+   ```bash
+   npm i -g @angular/cli
```
[the remark this answers is not in the document]
Comment (murphp15) [reply projects/frontend-ui/shared/projects/gui/README.md:9]: I think that we shouldn't specify the version as they will have other projects using this package too. 
Given that we are expecting them to use the global angular/cli package we should work with as many versions as possible. 
We would want to know asap if it doesn't work with a newer version or any version ever.

File: projects/frontend-ui/shared-components/projects/gui/README.md (hunk 11 above)
Comment (murphp15) [reply projects/frontend-ui/shared-components/projects/gui/README.md]: I think that we shouldn't specify the version as they will have other projects using this package too.
Given that we are expecting them to use the global angular/cli package we should work with as many versions as possible.
We would want to know asap if it doesn't work with a newer version or any version ever.

File: projects/frontend-ui/shared/projects/gui/README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (ivakoleva) [reply projects/frontend-ui/shared/projects/gui/README.md:9]: I agree. Looks optimistic, lets see how our cicd introduction works :)

File: projects/frontend-ui/shared-components/projects/gui/README.md (hunk 4 above)
Comment (ivakoleva) [reply projects/frontend-ui/shared-components/projects/gui/README.md]: there is still documentation-ui dir structure

File: specs/vep-1507-vdk-operations-ui/README.md [hunk 18]
```diff
@@ -15,39 +15,36 @@
 - [Alternatives](#alternatives)
 
 ## Summary
-
-<!--
-Short summary of the proposal. It will be used as user-focused
-documentation such as release notes or a (customer facing) development roadmap.
-The tone and content of the `Summary` section should be
-useful for a wide audience.
--->
```
Comment (ivakoleva) [inline specs/vep-1507-vdk-operations-ui/README.md:24]: nit: are we usually deleting the template section comments in early VEP stage? we may keep those to help with populating the VEP further

File: specs/vep-1507-vdk-operations-ui/README.md [hunk 19]
```diff
@@ -15,39 +15,36 @@
...
+Create an open source frontend for the VDK.
```
Comment (ivakoleva) [inline specs/vep-1507-vdk-operations-ui/README.md:18]: nit: isn't everything in versatile-data-kit repo open source? seems a tautology

File: specs/vep-1507-vdk-operations-ui/README.md [hunk 20]
```diff
@@ -15,39 +15,36 @@
...
+The frontend will let customers, create,delete deploy and generally managed nad browse their jobs.
+For frontend is written in an extensible format to easily allow other teams to build extensions.
```
Comment (ivakoleva) [inline specs/vep-1507-vdk-operations-ui/README.md:21]: I am missing more information about this frontend proposal. We have a sever-client model used for data jobs management, right? What types of clients do we have currently, and why do we need another client, presumably the frontend is a client?

File: specs/vep-1507-vdk-operations-ui/README.md [hunk 21]
```diff
@@ -15,39 +15,36 @@
...
-Non-goals are only features, which a contributor can reasonably assume were a goal.
-One example is features that were cut during scoping.
--->
+1. website should be intuitive, fast, responsive
+2. It should be easy to setup and run locally
+3. It should be easy to extend with new features
+4. It should be well tested
+5. It should be well documented
 
 ## High-level design
```
Comment (ivakoleva) [inline specs/vep-1507-vdk-operations-ui/README.md:41]: Where does the frontend proposed live, and how is it being used by the end customer?

**Review by ivakoleva — CHANGES_REQUESTED** [review]
VEP LGTM

**Comment by ivakoleva** [comment]
LGTM overall, I have some comments mainly on extending the VEP, and some cleanup

File: specs/vep-1507-vdk-operations-ui/README.md [hunk 22]
```diff
@@ -15,39 +15,36 @@
...
-<!--
-It tells **what** is it trying to achieve?
-List the specific goals (functional and nonfunctional requirements)? How will we
-know that this has succeeded?
 
-Specify non-goals. Clearly, the list of non-goals can't be exhaustive.
-Non-goals are only features, which a contributor can reasonably assume were a goal.
-One example is features that were cut during scoping.
--->
+1. website should be intuitive, fast, responsive
```
Comment (tozka) [inline specs/vep-1507-vdk-operations-ui/README.md:34]: We should avoid such unspecific requirments. What does a fast mean? or easy? 

They should be specific/concrete. Though if we currently do not information , we can leave a TODO for later.

File: specs/vep-1507-vdk-operations-ui/README.md [hunk 23]
```diff
@@ -15,39 +15,36 @@
...
 
 ## Motivation
-<!--
-It tells **why** do we need X?
-Describe why the change is important and the benefits to users.
-Explain the user problem that need to be solved.
--->
+1. Improve usability
+   1. With a UI customers will be able to use VDK much quicker and much more proficiently because UIs are easier to understand and become familiar with than CLI tools
+2. increase VDK adoption
```
Comment (tozka) [inline specs/vep-1507-vdk-operations-ui/README.md:29]: I think the motivation section should be from the point of view of the user. It's written in the https://github.com/vmware/versatile-data-kit/edit/main/specs/NNNN-template/README.md: 

```
Describe why the change is important and the benefits to users.
Explain the user problem that need to be solved [not your problem]
```

File: specs/vep-1507-vdk-operations-ui/README.md (hunk 19 above)
Comment (tozka) [inline specs/vep-1507-vdk-operations-ui/README.md:18]: Please split out the VEP in a separate PR. 

We want to break code commits into small self-contained units which can be merged quickly and independently.

File: specs/NNNN-template/README.md (hunk 9 above)
Comment (murphp15) [reply specs/NNNN-template/README.md:12]: This comment was redundant because we always need 2 approvers now so this doesn't need to be explicit.

File: projects/frontend-ui/shared-components/projects/gui/README.md (hunk 4 above)
Comment (ivakoleva) [reply projects/frontend-ui/shared-components/projects/gui/README.md]: there is still documentation-ui dir structure

File: specs/NNNN-template/README.md (hunk 9 above)
Comment (ivakoleva) [reply specs/NNNN-template/README.md:12]: please create a separate PR for such changes that are not related to the frontend change?

File: projects/frontend-ui/shared/CHANGELOG.md (hunk 13 above)
Comment (murphp15) [reply projects/frontend-ui/shared/CHANGELOG.md:2]: Yeah there is a CHANGELOG.md for every projects (control-service, core etc..)? 
Where does it get dynamically generated for other projects?

File: specs/NNNN-template/README.md (hunk 9 above)
Comment (murphp15) [reply specs/NNNN-template/README.md:12]: Cool

File: projects/frontend-ui/shared-components/projects/gui/README.md (hunk 4 above)
Comment (murphp15) [reply projects/frontend-ui/shared-components/projects/gui/README.md]: in what file do you see it referenced?

File: specs/vep-1507-vdk-operations-ui/README.md (hunk 8 above)
Comment (murphp15) [reply specs/vep-1507-vdk-operations-ui/README.md]: yeah it was like that before. 
I will revert.

**Comment by murphp15** [comment]
@ivakoleva  yeah, should be closed will do that now.
